cucumber-core/src/main/java/io/cucumber/core/options/CucumberPropertiesParser.java

```diff
@@ -195,10 +203,29 @@ private static <T> Function<String, Collection<T>> splitAndMap(Function<String,
                 .collect(toList());
     }
 
-    private static Stream<Collection<FeatureWithLines>> parseRerunFile(String property) {
+    private static Set<String> listRerunFiles(String path) {
```

Comment:
Consider returning a `Set<Path>` instead. The file visitor already provides you with `Path`s and the `parseFeatureWithLinesFile` also consumes paths.

Reply:
And to make reading this code a bit easier, putting this method after `parseRerunFiles` would be nice.